### PR TITLE
dockerfile-language-server-nodejs: fix build

### PR DESCRIPTION
**Pull request by Net-Mist**
## Description of changes

When trying to use the dockerfile-language-server, it fails to start with message "Error: Cannot find module '../lib/server".

In package.json there is an additional step to generate this file : `"prepublishOnly": "tsc -p ./src"`.

I don't know a lot JS, so their might be a clever way to fix this issue, but adding `tsc -p ./src` to the build command solve it for me (tested with Helix editor)
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Net-Mist** [comment]
Result of `nixpkgs-review pr 255259` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dockerfile-language-server-nodejs</li>
  </ul>
</details>

File: pkgs/development/tools/language-servers/dockerfile-language-server-nodejs/default.nix [hunk 1]
```diff
@@ -14,6 +14,13 @@ buildNpmPackage rec {
     hash = "sha256-uwwwF1eMoSA2C5h56BBllTZW8zRHueNeVwhwtycrNfA=";
   };
 
+  # patch build script to also run `tsc -p ./src`
+  postPatch = ''
+      substituteInPlace "./package.json" --replace \
+        "tsc -p ." \
+        "tsc -p ./src && tsc -p ."
+    '';
```
Comment (dotlambda) [inline pkgs/development/tools/language-servers/dockerfile-language-server-nodejs/default.nix:22]:
```suggestion
  preBuild = ''
      npm run prepublishOnly
    '';
```

**Review by dotlambda — COMMENTED** [review]
Please remove `nodePackages.` from the commit message.

**Comment by dotlambda** [comment]
@Net-Mist Do you want to add yourself as maintainer?

File: pkgs/development/tools/language-servers/dockerfile-language-server-nodejs/default.nix (hunk 1 above)
Comment (Net-Mist) [reply pkgs/development/tools/language-servers/dockerfile-language-server-nodejs/default.nix]: thanks a lot for the suggestion !

**Comment by Net-Mist** [comment]
> @Net-Mist Do you want to add yourself as maintainer?

@dotlambda with pleasure!

File: pkgs/development/tools/language-servers/dockerfile-language-server-nodejs/default.nix (hunk 1 above)
Comment (Net-Mist) [reply pkgs/development/tools/language-servers/dockerfile-language-server-nodejs/default.nix]: fixed

**Comment by erdnaxe** [comment]
LGTM, maybe split the commit adding the new maintainer? Quoting CONTRIBUTING.md:
> When adding yourself as maintainer, in the same pull request, make a separate commit with the message maintainers: add `<handle>`. Add the commit before those making changes to the package or module. See [Nixpkgs Maintainers](https://github.com/NixOS/nixpkgs/blob/maintainers/README.md) for details.

---

Result of `nixpkgs-review pr 255259` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dockerfile-language-server-nodejs</li>
  </ul>
</details>

**Comment by Net-Mist** [comment]
@erdnaxe thanks for the tip. I've split the commit into 2

**Comment by dotlambda** [comment]
> maintainers: add `<handle>`

That's for when you add yourself to `maintainers/maintainer-list.nix`

**Comment by Net-Mist** [comment]
@dotlambda So should I revert to a single commit ?

**Comment by erdnaxe** [comment]
Arg, my bad. Thanks for correcting me @dotlambda and sorry @Net-Mist !

**Comment by Net-Mist** [comment]
@erdnaxe no problem,
revert to a single commit

**Comment by Net-Mist** [comment]
Do you think I should also add 
```nix
passthru.updateScript = nix-update-script { };
```

**Comment by dotlambda** [comment]
> Do you think I should also add
> 
> ```nix
> passthru.updateScript = nix-update-script { };
> ```

I don't think that's needed. `nix-update` should work out of the box.

**Pull request merged**